src/sage/env.py

```diff
@@ -199,8 +199,29 @@ def var(key: str, *fallbacks: Optional[str], force: bool = False) -> Optional[st
 GRAPHS_DATA_DIR = var("GRAPHS_DATA_DIR", join(SAGE_SHARE, "graphs"))
 ELLCURVE_DATA_DIR = var("ELLCURVE_DATA_DIR", join(SAGE_SHARE, "ellcurves"))
 POLYTOPE_DATA_DIR = var("POLYTOPE_DATA_DIR", join(SAGE_SHARE, "reflexive_polytopes"))
-GAP_LIB_DIR = var("GAP_LIB_DIR", join(SAGE_LOCAL, "lib", "gap"))
-GAP_SHARE_DIR = var("GAP_SHARE_DIR", join(SAGE_SHARE, "gap"))
+
+# When the GAP SPKG is used, the corresponding variable in sage-conf
+# will be empty, so the final result will look like
+#
+#   GAP_ROOT_PATHS="$SAGE_LOCAL/lib/gap;$SAGE_SHARE/gap;"
+#
+# When the system GAP is used, on the other hand, we'll have e.g.
+#
+#   GAP_ROOT_PATHS="$SAGE_LOCAL/lib/gap;$SAGE_SHARE/gap;/usr/lib/gap"
+#
+# And both of these are correct enough. Suffixing the list with a
+# semicolon is allowed and in fact tells GAP to prepend to the list of
+# root paths rather than replace it. This is quite fine in the SPKG
+# case because we'll wind up with the same list of root paths being
+# included twice. It's also fine for the system GAP because we want
+# sage's GAP packages to take precedence (just like with executables
+# in the usual $PATH) over the system ones. In any case, the only
+# thing to watch out for here is the possibility of the empty string
+# after calling split(";").
+GAP_ROOT_PATHS = ";".join([join(SAGE_LOCAL, "lib", "gap"),
+                           join(SAGE_SHARE, "gap"),
+                           var("GAP_ROOT_PATHS", "")])
```

Comment:
This seems to make it impossible to really override the value of `GAP_ROOT_PATHS`. If either `SAGE_LOCAL/lib/gap` or `SAGE_SHARE/gap` is undesired for some reason, they are nevertheless forced.

Reply:
Also, can we do `SAGE_LOCAL/share/gap` instead of `SAGE_SHARE/gap`? For a regular installation of sage these two are the same, but for a system wide gap they are not.

For instance, I define `SAGE_SHARE` to be `/usr/share/sagemath/` and I place *sagemath specific* data inside (e.g. all databases that are only used from sage). And for a system sagemath we possibly have `SAGE_LOCAL=/usr` (taken from `SAGE_VENV` which itself defaults to `sys.prefix`)

(ideally, `SAGE_SHARE` can be phased out as in `conway_polynomials`, etc.)

Reply:
I do have a similar setup for `SAGE_SHARE`, I am hoping it is on its way out in the near future but yes, I do a similar distinction and patching of some paths for it. So, I second the request of @tornaria

Reply:
> This seems to make it impossible to really override the value of `GAP_ROOT_PATHS`. If either `SAGE_LOCAL/lib/gap` or `SAGE_SHARE/gap` is undesired for some reason, they are nevertheless forced.

Done. I moved the hackiness out of sage.env and into sage-conf.